src/scheduler/job_info.c

```diff
@@ -3426,11 +3429,42 @@ find_jobs_to_preempt(status *policy, resource_resv *hjob, server_info *sinfo, in
 
 		update_universe_on_end(npolicy, pjob,  "S", NO_ALLPART);
 		rjobs_count--;
+		/* Check if any of the previously preempted job increased its preemption priority */
+		for (ind = 0; pjobs[ind] != NULL; ind++) {
+			if (pjobs[ind]->job->preempt >= nhjob->job->preempt) {
+				dont_preempt_job = 1;
```

Comment:
not sure that I understand this, why don't we want to preempt 'pjob' if any of the previously preempted jobs increased their p-priority? What has other jobs' p-priority got to do with pjob?

Reply:
We don't want to preempt pjob because if we preempt this job then it increases the preemption priority of one of the previously preempted job. For example, jobs j1 runs with 8 ncpus and it is over the limit by 12 ncpus. So preempting J1 would still mean that the job is over its soft limit. Now comes job j2 which is consuming 6 ncpus and preempting this job will mean that previously preempted job J1 would now be under the softlimit and change its preemption priority.
Now we don't want to rollback the calculations we have done to preempt J1 and jobs after that so the easiest thing to do is to rollback J2 and not preempt it.

Reply:
Now that I look at this again, I think you need to change '>=' to just '>'.  It isn't a problem that we preempt a job and the preemptee is equal to the high priority job after it is preempted.  Jobs at equal preemption priority can't preempt each other.  It's also wrong to allow an entity to become immune to preemption just because they are over their limit by one.  Until they are at their limit, they are preemptable.